cumulus/parachains/runtimes/assets/common/src/erc721_transactor.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The ERC721 Asset Transactor.
+
+use core::marker::PhantomData;
+use ethereum_standards::IERC721;
+use frame_support::traits::{fungible::Inspect, OriginTrait};
+use pallet_revive::{
+	precompiles::alloy::{
+		primitives::{Address},
+		sol_types::SolCall,
+	},
+	AddressMapper, ContractResult, DepositLimit, MomentOf,
+};
+use sp_core::{Get, H160, H256, U256}; // tieni questo per U256::zero()
+
+use sp_runtime::Weight;
+use xcm::latest::prelude::*;
+use xcm_executor::{
+	traits::{ConvertLocation, Error as MatchError, MatchesNonFungibles, TransactAsset},
+	AssetsInHolding,
+};
+type BalanceOf<T> = <<T as pallet_revive::Config>::Currency as Inspect<
+	<T as frame_system::Config>::AccountId,
+>>::Balance;
+
+use pallet_revive::precompiles::alloy::primitives::U256 as AlloyU256;
+
+/// An Asset Transactor that deals with ERC721 tokens.
+fn sp_u256_to_alloy(x: U256) -> AlloyU256 {
+	let bytes: [u8; 32] = x.to_big_endian();
+	AlloyU256::from_be_bytes(bytes)
+}
```

Comment:
I didn't know this had to be done. Isn't it possible to do `into()`? Maybe it's not implemented